src/libs/actions/Report.js

```diff
@@ -158,6 +158,97 @@ function getSimplifiedReportObject(report) {
     };
 }
 
+/**
+ * Get a simplified version of an IOU report
+ *
+ * @param {Object} reportData
+ * @param {Number} reportData.transactionID
+ * @param {Number} reportData.amount
+ * @param {String} reportData.currency
+ * @param {String} reportData.created
+ * @param {String} reportData.comment
+ * @param {Object[]} reportData.transactionList
+ * @param {String} reportData.ownerEmail
+ * @param {String} reportData.managerEmail
+ * @param {Number} reportData.reportID
+ * @returns {Object}
+ */
+function getSimplifiedIOUReport(reportData) {
+    const transactions = _.map(reportData.transactionList, transaction => ({
+        transactionID: transaction.transactionID,
+        amount: transaction.amount,
+        currency: transaction.currency,
+        created: transaction.created,
+        comment: transaction.comment,
+    }));
+
+    return {
+        reportID: reportData.reportID,
+        ownerEmail: reportData.ownerEmail,
+        managerEmail: reportData.managerEmail,
+        currency: reportData.currency,
+        transactions,
+    };
+}
+
+/**
+ * Fetches the updated data for an IOU Report and updates the IOU collection in Onyx
+ *
+ * @param {Object} report
+ * @param {String[]} participants
+ */
+function updateIOUReportData(report, participants) {
+    const reportActionList = lodashGet(report, ['reportActionList'], []);
+    const containsIOUAction = _.any(reportActionList, reportAction => reportAction.action === 'IOU');
+
+    // If there aren't any IOU actions, we don't need to fetch any additional data
+    if (!containsIOUAction) {
+        return;
+    }
+
+    // If we don't have one participant, this is not an IOU
+    if (participants.length !== 1) {
+        return;
+    }
+
+    // Since the Chat and the IOU are different reports with different reportIDs, and GetIOUReport only returns the
+    // IOU's reportID, keep track of the IOU's reportID so we can use it to get the IOUReport data via `GetReportStuff`
+    let iouReportID = 0;
+    API.GetIOUReport({
+        debtorEmail: participants[0],
```

Comment:
How do you know that the first participant is the other user and not the current user?

Reply:
There will only be one participant here, since `getParticipantEmailsFromReport` calls `_without` and removes the currentUser.

Reply:
👍